feature-libs/asm/components/customer-list/customer-list.component.ts

```diff
@@ -0,0 +1,264 @@
+        currentPage: this.currentPage,
+      };
+      if (this.sortCode) {
+        options.sort = this.sortCode;
+      }
+
+      this.asmService.customerListCustomersSearchReset();
+
+      this.asmService.customerListCustomersSearch(options);
+    }
+  }
+
+  onChangeCustomerGroup(): void {
+    this.currentPage = 0;
+    this.sorts = null;
+    this.sortCode = '';
+    this.fetchCustomers();
+  }
+
+  getGroupName(
+    customerListsPage: CustomerListsPage,
+    id: string | undefined
+  ): string {
+    return (
+      customerListsPage?.userGroups?.find((userGroup) => userGroup.uid === id)
+        ?.name ?? ''
+    );
+  }
+
+  getBadgeText(customerEntry: User): string {
+    return (
+      (customerEntry.firstName?.charAt(0) ?? '') +
+      (customerEntry.lastName?.charAt(0) ?? '')
+    );
+  }
+
+  startColumnAction(
+    customerEntry: User,
+    action: CustomerListColumnActionType
+  ): void {
+    this.selectedCustomer = customerEntry;
+    let closeValue: CustomerListAction = {
+      actionType: action,
+      selectedUser: customerEntry,
+    };
+    this.closeModal(closeValue);
+  }
+
+  changeSortCode(sortCode: string): void {
+    this.sortCode = sortCode;
+    this.fetchCustomers();
+  }
+
+  goToNextPage(): void {
+    if (this.currentPage >= this.maxPage) {
+      this.currentPage = this.maxPage;
+    } else {
+      if (this.loaded) {
+        this.currentPage++;
+        this.fetchCustomers();
+      }
+    }
+  }
+
+  goToPreviousPage(): void {
+    if (this.currentPage <= 0) {
+      this.currentPage = 0;
+    } else {
+      if (this.loaded) {
+        this.currentPage--;
+        this.fetchCustomers();
+      }
+    }
+  }
+
+  closeModal(reason?: any): void {
+    this.launchDialogService.closeDialog(reason);
+  }
+
+  getSortLabels(): Observable<{
+    byNameAsc: string;
+    byNameDesc: string;
+    byOrderDateAsc: string;
+    byOrderDateDesc: string;
+    byDateAsc: string;
+    byDateDesc: string;
+  }> {
+    return combineLatest([
+      this.translation.translate('asm.customerList.tableSort.byNameAsc'),
+      this.translation.translate('asm.customerList.tableSort.byNameDesc'),
+      this.translation.translate('asm.customerList.tableSort.byDateAsc'),
+      this.translation.translate('asm.customerList.tableSort.byDateDesc'),
+      this.translation.translate('asm.customerList.tableSort.byOrderDateAsc'),
+      this.translation.translate('asm.customerList.tableSort.byOrderDateDesc'),
+    ]).pipe(
+      map(
+        ([
+          textByNameAsc,
+          textByNameDesc,
+          textByOrderDateAsc,
+          textByOrderDateDesc,
+          textByDateAsc,
+          textByDateDesc,
+        ]) => {
+          return {
+            byNameAsc: textByNameAsc,
+            byNameDesc: textByNameDesc,
+            byOrderDateAsc: textByOrderDateAsc,
+            byOrderDateDesc: textByOrderDateDesc,
+            byDateAsc: textByDateAsc,
+            byDateDesc: textByDateDesc,
+          };
+        }
```

Comment:
I'm surprised prettier did not truncate ths

```suggestion
        ]) => ({
          byNameAsc: textByNameAsc,
          byNameDesc: textByNameDesc,
          byOrderDateAsc: textByOrderDateAsc,
          byOrderDateDesc: textByOrderDateDesc,
          byDateAsc: textByDateAsc,
          byDateDesc: textByDateDesc,
        })
```

Reply:
we will update in ASM 360